consensus/babe/runtime.go

```diff
@@ -93,12 +114,15 @@ func (b *Session) finalizeBlock() (*types.Block, error) {
 }
 
 // calls runtime API function TaggedTransactionQueue_validate_transaction
-func (b *Session) validateTransaction(e types.Extrinsic) (*tx.Validity, error) {
-	// TODO: use allocator to store extrinsic
-	var loc int32 = 1000
-	b.rt.Store(e, loc)
+func (b *Session) validateTransaction(data types.Extrinsic) (*tx.Validity, error) {
+	ptr, err := b.rt.Malloc(uint32(len(data)))
+	if err != nil {
+		return nil, err
+	}
+
+	b.rt.Store(data, int32(ptr))
 
-	ret, err := b.rt.Exec(runtime.TaggedTransactionQueueValidateTransaction, loc, e)
+	ret, err := b.rt.Exec(runtime.TaggedTransactionQueueValidateTransaction, int32(ptr), data)
```

Comment:
Is `ret` short for return? We should be more descriptive here and elsewhere (ie, `rawdata`).